models/certs.go

```diff
@@ -32,3 +34,51 @@ type CertInfo struct {
 	ValidTo   string    `json:"ValidTo"`
```

Comment:
Based on the provided diff, the following improvements can be suggested:

**Unused imports**: The `package models` line seems to be unused. If it's not needed, it can be removed to keep the code clean and focused.

**Naming conventions**: The naming convention used for the `NewCSRConfigModel` and `NewCertInfoModel` functions is inconsistent with the rest of the code. It's recommended to follow the conventional naming scheme used in the codebase, which seems to be camelCase for functions and PascalCase for types.

**Magic strings**: In the `NewCertInfoModel` function, the "dummy" string is used as a default value for the `ID` field. It's recommended to replace this with a constant or a more robust way of replacing missing values.

**Types**: The `types` package is imported, but the exact version is not specified. It's recommended to use a specific version or keep the import statement consistent across the codebase.

Here's the updated code snippet with the recommended improvements:

@@ -13,6 +13,8 @@ limitations under the License.

package models

+import (
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)

type CSRConfig struct {
	DistinguishedName string `json:"DistinguishedName"`
	// ...
}

type CSRConfigModel struct {
	DistinguishedName types.String `tfsdk:"distinguished_name"`
	// ...
}

type CertInfoModel struct {
	ID        types.String   `tfsdk:"id"`
	IssuedTo  CSRConfigModel `tfsdk:"issued_to"`
	IssuedBy  CSRConfigModel `tfsdk:"issued_by"`
	// ...
}

func NewCSRConfigModel(input CSRConfig) CSRConfigModel {
	return CSRConfigModel{
		DistinguishedName: types.StringValue(input.DistinguishedName),
		// ...
	}
}

func NewCertInfoModel(info CertInfo) CertInfoModel {
	const defaultID = "dummy"
	return CertInfoModel{
		ID:        types.StringValue(defaultID),
		IssuedTo:  NewCSRConfigModel(info.IssuedBy),
		IssuedBy:  NewCSRConfigModel(info.IssuedTo),
		// ...
	}
}